src/python/pants/engine/internals/platform_rules_test.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import annotations
+
+from textwrap import dedent
+
+from pants.core.util_rules.environments import DockerEnvironmentTarget
+from pants.engine.env_vars import CompleteEnvironmentVars
+from pants.engine.environment import EnvironmentName
+from pants.testutil.rule_runner import QueryRule, RuleRunner
+
+
+def test_docker_complete_env_vars() -> None:
+    rule_runner = RuleRunner(
+        rules=[QueryRule(CompleteEnvironmentVars, [])],
+        target_types=[DockerEnvironmentTarget],
+        singleton_environment=EnvironmentName("docker"),
+    )
+    rule_runner.write_files(
+        {
+            "BUILD": dedent(
+                """\
+                _docker_environment(
+                    name='docker',
+                    image='centos@sha256:a1801b843b1bfaf77c501e7a6d3f709401a1e0c83863037fa3aab063a7fdb9dc',
```

Comment:
To reduce the total amount of docker image pulling that we need to do in CI, we should probably try to keep most of our tests on a small set of images. Having good hygiene from the start would help.

For now, maybe move this image name into a constant somewhere in a testutil file, so that we can be thinking about the total number / shape of images as we go?

Reply:
I'm not sure yet where this should live. I plan to write more tests for Docker this week, so I'll wait.